e2e/testcases/hydration_test.go

```diff
@@ -52,6 +54,29 @@ func TestHydrateKustomizeComponents(t *testing.T) {
 		ntopts.Unstructured,
 	)
 
+	// Dry configs not yet added to repo, assert that hydration is disabled
```

Comment:
Is it necessary to assert hydration controller disabled  later in the same test, when DRY configs are removed and hydration controller was known enabled?

Reply:
Just to be clear, the hydration controller is:
- enabled when dry configs are present
- disabled when dry configs are absent (i.e. only wet configs)

That being said, these before/after assertions are not strictly necessary but accomplish a few things:
- Verify the assumptions/correctness of the reconciler/reconciler-manager logic that is dynamically toggling the hydration-controller
- Provides a clear signal in the case of failure that the reason the test failed is because of a bug in the aforementioned logic